cargo-dist/templates/ci/github_ci.yml.j2

```diff
@@ -85,7 +87,7 @@ jobs:
   upload-local-artifacts:
     # Let the initial task tell us to not run (currently very blunt)
     needs: plan
-    if: ${{ fromJson(needs.plan.outputs.val).releases != null }}
+    if: ${{ fromJson(needs.plan.outputs.val).releases != null && fromJson(needs.plan.outputs.val).ci.github.pr_run_mode == "upload" }}
```

Comment:
this isn't quite right, this will make pr_run_mode = "skip" prevent releases altogether. needs to be combine with whether we're releasing for realz.